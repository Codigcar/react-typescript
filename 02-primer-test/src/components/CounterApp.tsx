import React, { useState } from 'react'

export const CounterApp = () => {
    const [contador, setContador] = useState(0)

    const handleSuma = () => {
        setContador(contador + 1);
    }

    const handleResta = () => {
        setContador(contador - 1);
    }

    return (
        <>
            <div className="container">
                <h2>Contador: {contador}</h2>
                <button onClick={handleSuma} className="btn btn-primary">+1</button>
                <button onClick={handleResta} className="btn btn-danger">-1</button>
            </div>
        </>
    )
}
