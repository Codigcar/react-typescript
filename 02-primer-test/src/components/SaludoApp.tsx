
interface propsInterface {
    saludo:string;
    subtitulo:string;
}

export const SaludoApp = (props:propsInterface) => {
    return (
        <>
        <div className="container">
            <h1>{props.saludo}</h1>
            <p>{props.subtitulo}</p>
        </div>
        </>
    )
}
