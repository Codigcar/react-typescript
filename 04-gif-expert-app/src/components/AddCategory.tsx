import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = (props:any) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e: any) => {
    setInputValue(e.target.value);
};

const handleSubmit = (e: any) => {
    e.preventDefault();
    if(inputValue.trim().length > 2){
      props.setCategories((categories: any) => [inputValue, ...categories,]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        value={inputValue}
        onChange={handleInputChange}
        type="text"
        className="input-text"
      />
    </form>
  );
};

AddCategory.prototype = {
  props: PropTypes.func.isRequired
}