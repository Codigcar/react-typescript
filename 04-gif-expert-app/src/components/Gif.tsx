import React from 'react'
import PropTypes from 'prop-types';

export const Gif = (props:any) => {
    return (
        <div className="card">
            <img src={props.url} alt={props.title} />
            <p>{props.title}</p>
        </div>
    )
}
/* export const Gif = ({title, url}:any) => {
    return (
        <div className="card">
            <img src={url} alt="props.title" />
            <p>{title}</p>
        </div>
    )
} */

Gif.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}
