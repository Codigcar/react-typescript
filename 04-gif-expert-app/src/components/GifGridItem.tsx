import React from 'react'

interface propsInterface {
    category:string;
}
type imgType = {
    id: string;
    title: string;
    images: { downsized_medium: { url: string; }; }; 
}

export const GifGridItem = (props:propsInterface) => {

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=vfV8SOvefLtVRj5pOOTcRZgepQ0zs29o';
        const resp = await fetch(url);
        const {data} = await resp.json();

        const gif = data.map((img: imgType) => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gif);
        
    }
    getGifs();
    return (
        <div>
            {props.category}
        </div>
    )
}
