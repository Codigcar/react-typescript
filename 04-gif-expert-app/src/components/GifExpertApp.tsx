import React from "react";
import { useState } from "react";
import { AddCategory } from "./AddCategory";

export const GifExpertApp = () => {
  // const categories = ['category1','category2','category3'];
  const [categories, setCategories] = useState([
    "category1",
    "category2",
    "category3",
  ]);

  return (
    <>
      <h1>Encuentra tus gifs </h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>  
        {categories.map((categories, i) => {
          return <li key={i}>{categories}</li>;
        })}
      </ol>
    </>
  );
};
