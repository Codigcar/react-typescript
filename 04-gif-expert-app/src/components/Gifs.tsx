import React from 'react'
import { useState } from 'react';
import { useEffect } from 'react';
import { Gif } from './Gif';

interface propsInterface {
    category:string;
}
type gifType = {
    id: string;
    title: string;
    images: { downsized_medium: { url: string; }; }; 
}

export const Gifs = (props:propsInterface) => {

    const [gifs, setGifs] = useState([]);

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=vfV8SOvefLtVRj5pOOTcRZgepQ0zs29o';
        const resp = await fetch(url);
        const {data} = await resp.json();

        const arrayGifs = data.map((gif: gifType) => {
            return {
                id: gif.id,
                title: gif.title,
                url: gif.images?.downsized_medium.url
            }
        })
        console.log(arrayGifs);
        setGifs(arrayGifs);
    }

    useEffect(() => {
        getGifs();
    }, [])

    return (
        <div>
            <h3>{props.category}</h3>
            <ol>
                {
                    /* gifs.map((gif:gifType) => (
                        <li key={gif.id}>{ gif.title }</li>
                    )) */
                    gifs.map((gif:gifType) => (
                        <Gif key={gif.id} {...gif} />
                    ))
                }
            </ol>
        </div>
    )
}
