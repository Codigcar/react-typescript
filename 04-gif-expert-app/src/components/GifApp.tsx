import React from "react";
import { useState } from "react";
import { AddCategory } from "./AddCategory";
import { Gifs } from "./Gifs";

export const GifExpertApp = () => {
  // const categories = ['category1','category2','category3'];
  const [categories, setCategories] = useState(["category1"]);

  return (
    <>
      <h1>Encuentra tus gifs </h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>  
        {
            categories.map((category, i) => (
            //   return <li key={i}>{categories}</li>;
                <Gifs key={i} category={category} />
            ))
        }
      </ol>
    </>
  );
};
