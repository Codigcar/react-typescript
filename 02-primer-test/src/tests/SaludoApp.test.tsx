import 'jest';
import React from 'react';
import { shallow } from 'enzyme';
import { SaludoApp } from '../components/SaludoApp';
import { render } from '@testing-library/react';

describe('Pruebas en <SaludoApp/>', () => {
    test('debe de mostrar el mensaje "Hola, Soy Carlos"', () => {
        // const saludo = 'Carlos';
        
        const wrapper = shallow( <SaludoApp saludo='Carlos' subtitulo='Castilla' />)
        expect(wrapper).toMatchSnapshot();
        // const {getByText} = render(<SaludoApp saludo={saludo} subtitulo='Castilla'/>);
        // expect(getByText(saludo)).toBeInTheDocument();

    })
    
})
