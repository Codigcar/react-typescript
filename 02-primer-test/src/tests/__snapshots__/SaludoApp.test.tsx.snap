// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SaludoApp/> debe de mostrar el mensaje "Hola, Soy Carlos" 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h1>
      Carlos
    </h1>
    <p>
      Castilla
    </p>
  </div>
</Fragment>
`;
